### ISSUE TITLE
String `format` support

### ISSUE THREAD

**Issue by notpushkin** [adrienbrault/json-schema-to-gbnf#3]
Hey, first of all thanks for your wonderful project!

Right now the only way to restrict a format of a string is with `pattern`, but JSON Schema also supports `format` and some [built-in formats][1] could be pretty useful to support.

For example, email is notoriously [hard to parse][2] using a regex, but a fairly clean BNF is defined in [RFC 5321][3] (which JSON Schema uses for the `email` format) and I think it should be pretty straightforward to port.

Do you think this could be something you could use? I can try and work out a PR if you'd like :-)

[1]: https://json-schema.org/understanding-json-schema/reference/string#built-in-formats
[2]: https://stackoverflow.com/a/201378
[3]: https://datatracker.ietf.org/doc/html/rfc5321#section-4.1.2

**Comment by adrienbrault** [comment]
Absolutely, good idea.

There's a few formats already implemented in llama cpp `json-schema-to-grammar.mjs`: https://github.com/ggerganov/llama.cpp/blob/2ee44c9a1865a928ccbbc16a2d7841d7513f31c1/examples/server/public/json-schema-to-grammar.mjs#L245-L252
